src/connection_string.ts

```diff
@@ -407,6 +407,12 @@ export function parseOptions(
       });
     }
 
+    if (isAws && mongoOptions.credentials.username && !mongoOptions.credentials.password) {
+      throw new MongoParseError(
```

Comment:
"ParseError" is not really the correct error type to use here, since it's not that we were unable to parse the options, it's that one of the expected options wasn't provided. This is an instance of a `MongoAPIError`, perhaps the more specific `MongoMissingCredentialsError`.

Reply:
Thanks for the context, was not aware of NODE-4015. Using `MongoMissingCredentialsError` as suggested, agree the description of this error seems to fit this scenario well: 

>  An error generated when the user fails to provide authentication credentials before attempting
    to connect to a mongo server instance.